src/stan/math/prim/mat/meta/index_type.hpp

```diff
@@ -21,6 +21,16 @@ namespace stan {
       typedef typename Eigen::Matrix<T,R,C>::Index type;
     };
 
+    template <typename T>
+    struct index_type<Eigen::MatrixBase<T> > {
+      typedef typename index_type<T>::type type;
+    };
+
+    template <typename T>
+    struct index_type<Eigen::Map<T> > {
```

Comment:
What were the compiler errors?

- Bob

> On Mar 12, 2015, at 8:43 AM, Joshua Pritikin <notifications@github.com> wrote:
> 
> In src/stan/math/prim/mat/meta/index_type.hpp:
> 
> > @@ -21,6 +21,16 @@ namespace stan {
> >        typedef typename Eigen::Matrix<T,R,C>::Index type;
> >      };
> >  
> > +    template <typename T>
> > +    struct index_type<Eigen::MatrixBase<T> > {
> > +      typedef typename index_type<T>::type type;
> > +    };
> > +
> > +    template <typename T>
> > +    struct index_type<Eigen::Map<T> > {
> 
> Hmmm.... does that mean we need to have separate function signatures for MatrixBase and Map?
> 
> In general, no, because MatrixBase can wrap Map. However, due to the way template specialization works, I had to add both signatures in some places to get things to compile. I certainly didn't want to add more than MatrixBase, but I felt that I didn't have a choice when the compiler complained. Maybe there is a better way to get things working.

What was the compiler error when you didn't have it?  

> 
> —
> Reply to this email directly or view it on GitHub.
>